Support go version 1.25

**Describe the bug**
With the newest upgrade for golang there is a problem with the golang.org experimental tools package. 

`pkg/mod/golang.org/x/tools@v0.24.0/internal/tokeninternal/tokeninternal.go:64:9: invalid array length -delta * delta (constant -256 of type int64)`

**To Reproduce**

`FROM public.ecr.aws/docker/library/golang:1.25.0
RUN go install github.com/Khan/genqlient@v0.8.1
`

**Expected behavior**
Its working without problems :D 

**genqlient version**
v0.8.1

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go: [ '1.22', '1.23' ]
+        go: [ '1.23', '1.24', '1.25' ]
 
     steps:
     - name: Set up Go

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -20,6 +20,10 @@ When releasing a new version:
 
 <!-- Add new changes in this section! -->
 
+This release adds support for Go 1.25.
+
+Note that genqlient now requires Go 1.23 or higher, and is tested through Go 1.23, 1.24 and 1.25.
+
 ### Breaking changes:
 
 ### New features:
@@ -57,7 +61,7 @@ Note that genqlient now requires Go 1.22.5 or higher, and is tested through Go 1
 
 ### New features:
 
-- genqlient now supports subscriptions; the websocket protocol is by default `graphql-transport-ws` but can be set to another value.  
+- genqlient now supports subscriptions; the websocket protocol is by default `graphql-transport-ws` but can be set to another value.
   See the [documentation](subscriptions.md) for more details on how to use subscriptions.
 - genqlient now supports double-star globs for schema and query files; see [`genqlient.yaml` docs](genqlient.yaml) for more.
 - genqlient now generates slices containing all enum values for each enum type.

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/Khan/genqlient
 
-go 1.22.5
+go 1.23.0
 
 toolchain go1.23.3
 
@@ -13,7 +13,7 @@ require (
 	github.com/gorilla/websocket v1.5.0
 	github.com/stretchr/testify v1.9.0
 	github.com/vektah/gqlparser/v2 v2.5.19
-	golang.org/x/tools v0.24.0
+	golang.org/x/tools v0.36.0
 	gopkg.in/yaml.v2 v2.4.0
 )
 
@@ -25,7 +25,7 @@ require (
 	github.com/hashicorp/golang-lru/v2 v2.0.7 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/sosodev/duration v1.3.1 // indirect
-	golang.org/x/mod v0.20.0 // indirect
-	golang.org/x/sync v0.8.0 // indirect
+	golang.org/x/mod v0.27.0 // indirect
+	golang.org/x/sync v0.16.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )

